akka-testkit/src/test/scala/akka/testkit/TestEventListenerSpec.scala

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.testkit
+
+import akka.event.Logging
+import akka.event.Logging.Error
+import akka.event.Logging.Warning
+
+class TestEventListenerSpec extends AkkaSpec with ImplicitSender {
+
+  "The untyped EventFilter.error" must {
+    "filter errors without cause" in {
+      val filter = EventFilter.error()
+      filter(errorNoCause) should ===(true)
+    }
+
+    "filter errors with cause" in {
+      val filter = EventFilter.error()
+      filter(errorWithCause(new AnError)) should ===(true)
+    }
+  }
+
+  "The typed EventFilter" must {
```

Comment:
Perhaps also add a testcase that `EventFilter[Error.NoCause]` has the 'old' behavior (only filtering errors that have no cause)?

Reply:
If this is the desired behavior, then the current fix should be reworked. I guess an explicit `Error.All` class should be introduced, and `ErrorFilter.error` should use that (as opposed to using `Error.NoCause` as it currently does). This boils down to decide if it's useful for the user to filter only errors that have no cause attached. So, is this what you'd like?

Reply:
Ah, indeed `EventFilter[Error.NoCause.type]` will now filter everything.

I think it would be a useful improvement to allow filtering on errors without a cause, but it's not essential for this PR.